lib/m3u8-parser.js

```diff
@@ -22,6 +27,26 @@ var m3u8parser = module.exports = function() {
       self.emit('item', line.trim());
     }
   }
+  function ytdlParse() {
+    var time = self.ytdlSavedLines[self.ytdlSavedLines.length - 2];
```

Comment:
We have to get the time from the line, to determine how long the m3u8 packet is. Is it 5 seconds? 4 seconds? This will automatically get it for us. For every m3u8 item that comes through, right above it in the array is how long that item is, determining how long we should time out before asking for more information.